src/angularjs/src/app/places/list/places-list.controller.js

```diff
@@ -34,13 +41,40 @@
             ctl.getPrev = getPrev;
             ctl.places = [];
 
-            ctl.filters = {};
+            ctl.neighborhoodFilter = null;
+
+            ctl.sortBy = sortingOptions[0]; // default to alphabetical order
+            ctl.sortingOptions = sortingOptions;
+
+            ctl.getPlaces = getPlaces;
 
             getPlaces();
+            loadOptions();
+            $scope.$watch(function(){return ctl.neighborhoodFilter;}, filterNeighborhood);
```

Comment:
According to [this discussion](http://stackoverflow.com/questions/25113884/should-angular-watch-be-removed-when-scope-destroyed), Angular should be cleaning up watchers on its own during the scope destruction.